src/icp_v2.cc

```diff
@@ -481,8 +480,18 @@ doV2Query(int fd, Ip::Address &from, char *buf, icp_common_t header)
     int rtt = 0;
     int src_rtt = 0;
     uint32_t flags = 0;
-    /* We have a valid packet */
-    char *url = buf + sizeof(icp_common_t) + sizeof(uint32_t);
+    /* Ensure QUERY has space for the 32-bit field and at least one URL byte. */
+    const auto hdr = sizeof(icp_common_t);
+    const auto qhdr = hdr + sizeof(uint32_t);
```

Comment:
This is a size. Please indicate that in the variable name.
```suggestion
    const auto queryHeaderSize = sizeof(icp_common_t) + sizeof(uint32_t);
```
related lines will need adjusting to match